### ISSUE TITLE
Interfaces required in undelying Scala types?

### ISSUE THREAD

**Issue by luisobo** [sangria-graphql/sangria#93]
Hi again :grin: 

I may be missing something here, but here is my current situation:
- I have a Customization abstract type and several subtypes. 
- I have to add an interface at the graphql level called Customizable with one method `customizations` that returns a list of Customizations. The whole customization service is decoupled from anything else, so it does not make sense for me to make my types (case class Article, etc) implement the customizable interface in the core of my application.

I want to do this:
```scala
  val CustomizableInterface = InterfaceType("Customizable", fields[RequestContext, Node](
    Field("customizations", ListType(CustomizationInterface), resolve = c => customizationsService.listCustomizationForNode(c.value.id, List())(c.ctx))
  ))
```

Basically, I pull the customizations from a separate service, not from the objects that are customizable, like Article.

Note how the `Val` type is `Node` and not `Customizable`.

Is there a way to make it work without having to stick the Customizable scala interface to all my scala types (Article in the example)?

Man, worst explanation ever, my brain is fried, let me know if it does not make sense and thank you again.

**Comment by OlegIlyenko** [comment]
I think I understood the setup, but it's not yet clear what the actual issue is. Does it compile and run in a form that you have shown in the code example?

Or is the issue in GraphQL schema definition itself? In this case you can try to use union types. So instead of defining an interface, you can define a `CustomizableUnion`. Here is an example of this feature: https://github.com/sangria-graphql/sangria/blob/master/src/test/scala/sangria/execution/UnionInterfaceSpec.scala#L26

**Comment by luisobo** [comment]
Sorry for dropping the ball. I came back to this issue and it's working okay now. I was experiencing some crash inside sangria, when it was trying to resolve a type but I can't reproduce it. In any case, it works now. I'm going to close this and I'd report back if I had any other problems. 

Thank you!
